UX article (#159)
MechanicalRock/mechanicalrock.github.io: _posts/2021-05-05-the-competitive-advantage-of-ux.md

At Mechanical Rock, we achieve this by bringing UX together with DevOps, which enables lean development because the problem space is understood, solutions are tested with users prior to the build and iterations are based on data. Moreover, designers and developers work together with a shared context reducing misunderstanding and errors. Whereas, without UX, products often do not meet stakeholder or user needs ([Barnaby Dracup](https://www.devopsonline.co.uk/integrating-ux-with-devops/)).. In 2020, failed development projects cost $260 billion ([Roberto Torres](https://www.ciodive.com/news/poor-software-quality-report-2020/593015/)).

Based on this, I believe value and subsqeuent competitive advantage is greatly enabled by understanding how to optimise the relationship between UX and DevOps.
<br /><br />

### 1. Discovery

At Mechanical Rock before we start a piece of design work, we run in depth discovery sessions. Developers, designers and the clients are involved. In these sessions, we understand the problem space, dive deeper into the current state and user journey. A shared understanding is fostered as both designers and developers have the same context of the problem space.

> Without this session, it is too easy to make assumptions and be driven by the client leading to poor delivery. For example, on a recent project, the client was adamant that the users felt fearful of artifical intelligence and machine learning; however, our user research suggested otherwsie, which opened up new opportunities and allowed us to deliver solutions driven by user data. This is an example, of how UX provides visibility into the problem space and user needs which leads to strong delivery.

Reviewer: ```suggestion
Without this session, it is too easy to make assumptions and be driven by the client leading to poor delivery. For example, on a recent project, the client was adamant that the users felt fearful of artificial intelligence and machine learning; however, our user research suggested otherwise, which opened up new opportunities and allowed us to deliver solutions driven by user data. This is an example, of how UX provides visibility into the problem space and user needs which leads to strong delivery.
```

> ![Cypress CodeBuild report](/img/Quickmail_Discovery.jpg)

Reviewer: ```suggestion
![Facilitating a discovery workshop](/img/Quickmail_Discovery.jpg)
```

Reviewer: rename the image to redact - references here need updating.

_Karen Cheng, Principal Design Consultant, running a discovery workshop with a client_

<br /><br />

### 2. Removal of Silos

DevOps is about removing waste to support the flow of value through a system. This is enabled by the removal of silos, feedback loops and a shared understanding. Value in DevOps concerns whether features add value or are waste. Waste being a redundant functionality that is not useful. The flow of value through a system is impeded by constraints such as silos.

> Within product teams, silos occur when different specialisations work separately. Teams work in a vacuum, doing only their part without wider considerations, constraints, ideas and critiques of the larger team. For example, when development is siloed from UX, designers do not have visibility into software constraints, and therefore the system in which a user experience operates in. Decisions are made with partial context; impeding this flow and therefore value. Considering UX offers a long term competitive advantage “it’s important to understand these relationships from a UX perspective because they illuminate how interactions between enterprise systems can exert significant impacts on the user experience”, ([Amy Chess](https://www.uxmatters.com/mt/archives/2019/03/platform-ux-domain-driven-design-concepts-in-enterprise-ux.php)). Removal of silos allows continuous feedback loops and a shared understanding to foster between designers and developers which removes constraints and enables flow. Without that it is no different to pre-DevOps days.

Reviewer: ```suggestion
Within product teams, silos occur when different specialisations work separately. Teams work in a vacuum, doing only their part without wider considerations, constraints, ideas and critiques of the larger team. For example, when development is siloed from UX, designers do not have visibility into software constraints, and therefore the system in which a user experience operates in. Decisions are made with partial context; impeding this flow and therefore value. Considering UX offers a long term competitive advantage: “it’s important to understand these relationships from a UX perspective because they illuminate how interactions between enterprise systems can exert significant impacts on the user experience”, ([Amy Chess](https://www.uxmatters.com/mt/archives/2019/03/platform-ux-domain-driven-design-concepts-in-enterprise-ux.php)). Removal of silos allows continuous feedback loops and a shared understanding to foster between designers and developers which removes constraints and enables flow. Without that it is no different to pre-DevOps days.
```

Reviewer: I don't know what this change was!  Or you've already made it....

> At Mechanical Rock, we work in cross-functional teams where designers work closely with developers, throughout the entire project, which fosters feedback loops and supports a shared understanding. Moreover, developers and designers are involved in all meetings and decisions, such as example mapping sessions.

Reviewer: ```suggestion
At Mechanical Rock, we work in cross-functional teams where designers work closely with developers, throughout the entire project, which fosters feedback loops and supports a shared understanding. Moreover, developers and designers are involved in all meetings and decisions, such as [example mapping](https://mechanicalrock.github.io/2020/05/18/effective-example-mapping.html) sessions.
```

Including designers in a VSM workshop provides them with greater context on the current system, the constraints, gaps and opportunities and how that is feeding into poor user experience.

Without this context, designers have an incomplete understanding about the current state and no shared understanding between designers and developers. This can quickly lead to miscommunication and assumptions along with potential re-work on designs. This is not an effective approach. Whereas a shared understanding as [Kai Brunner](https://www.wired.com/insights/2015/01/is-devops-driving-the-future-of-ux-design/discusses) states allows problems to be "solved much faster without the drag of a struggle".

An approach such as a VSM workshop is highly effective in fostering a shared understanding between designers, developers and clients. VSM provides a space for UX and DevOps to work together and this is "essential for DevOps teams to ensure they solve the right problems and are understanding the user’s real needs", ([Barnaby Dracup](https://www.devopsonline.co.uk/integrating-ux-with-devops/)). And this supports a long term competitive advantage.

> ![Cypress CodeBuild report](/img/Value_Stream_Mapping.jpg)

Reviewer: ```suggestion
![Photograph of a Value Stream Mapping Workshop](/img/Value_Stream_Mapping.jpg)
```

---
layout: post
title: The Competitive Advantage of UX
date: 2021-05-05
tags: ux devops design competitive advantage optimisation
author: Lucy Sharp

> image: /img/User_Journey_Map.jpeg

Reviewer: Broken link
